### ISSUE TITLE
Properly support Apple Silicon (M1) devices

### ISSUE THREAD

**Issue by DeMoorJasper** [parcel-bundler/source-map#41]
Ensure this package also runs on M1 Macs, besides this also build binaries for these devices and publish them in a new release.

> This is also affecting M1 Macs running node 15.3.0 or possibly linked to https://github.com/parcel-bundler/parcel/issues/5046

_Originally posted by @WA9ACE in https://github.com/parcel-bundler/source-map/issues/33#issuecomment-733251373_

TODO:
- [ ] Fix any issues with compatibility
- [ ] Setup CI Pipeline for automatically building universal mac binaries

**Comment by DeMoorJasper** [comment]
> @WA9ACE that's not good, unfortunately nobody on the Parcel core team currently has an M1 mac and Azure doesn't support it yet either... so we can't really test this at the moment. Also you should probably open a new issue for Apple-M1 support as it's not really the same as linux arm
>
> I do have one on order but it won't be delivered until somewhere mid-december, so until than I can't really debug and fix this.
>
> Either way Azure needs to start supporting Apple-M1 based CI pipelines before we can release any binaries.

Copied from #33

**Comment by devongovett** [comment]
I think node still doesn't have official Apple Silicon builds?

**Comment by DeMoorJasper** [comment]
@devongovett I don't think so, it's probably related to this issue https://github.com/parcel-bundler/parcel/issues/5046

**Comment by WA9ACE** [comment]
Correct, this is an arm node built from 15.3.0 which includes a patch from upstream V8 solving a previously blocking issue

Node Merged V8 apple arm macro: https://chromium-review.googlesource.com/c/v8/v8/+/2517689
Node M1 Support tracking issue: https://github.com/nodejs/build/issues/2474

Node does not, as of the time of this writing, provide officially supported builds of Node for M1 Macs.

Also just to clarify I don't want to sound like I'm demanding an open source project support brand new hardware immediately. I know things take time and effort, both of which have a corresponding monetary value and that it is completely unreasonable of anyone to ask an OSS project to support 2 week old hardware immediately.

To give some context on things I've tried, I have gatekeeper disabled, and signed the binary by hand myself with `codesign`. I can verify the artifact is arm with `lipo -info sourcemap.node` returning.

`Non-fat file: sourcemap.node is architecture: arm64`

**Comment by mischnic** [comment]
@WA9ACE Are you sure this is related to https://github.com/parcel-bundler/parcel/issues/5046 and therefore the codesign situation? If you get the error described in https://github.com/parcel-bundler/source-map/issues/33, then it's probably more related to ARM itself.

**Comment by DeMoorJasper** [comment]
Just tried on my M1 Macbook I just got and it works fine when using Rosetta, which is probably still the recommended way of running NodeJS at this point?

@WA9ACE If you can walk me through how to set this up I can probably debug it further

**Comment by DeMoorJasper** [comment]
Resolved by #52
